src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala

```diff
@@ -95,33 +99,45 @@ object MatchMaker {
     // 5. Create chunks of size 4 and size 2
     val chunks = randomPlayerList.splitAt(splitIndex) match {
       case (first, second) => {
-        val allChunks = first.grouped(4).toList ::: second.toList.reverse.grouped(4).toList
+        val allChunks = first.grouped(doublesSize).toList ::: second.toList.reverse.grouped(doublesSize).toList
         val remainder = allChunks
-          .filterNot(_.size == 4)
+          .filterNot(_.size == doublesSize)
           .flatten
-          .grouped(4)
+          .grouped(doublesSize)
           .toList
-        val (left, right) = (allChunks.filter(_.size == 4) ::: remainder).splitAt(numDoublesCourts)
-        left ::: right.flatten.grouped(2).toList
+        val (left, right) = (allChunks.filter(_.size == doublesSize) ::: remainder).splitAt(numDoublesCourts)
+        left ::: right.flatten.grouped(singlesSize).toList
       }
     }
 
     // 6. Map the chunks into matches
-    matches = chunks
-      .map { players => players.splitAt(players.size/2) }
+    val sortedCourtTypes = courtTypes
       .zipWithIndex
-      .map { case ((team1, team2), ind) =>
-        val courtId = ind + 1
+      .map { case (matchType, ind) =>
+        (ind + 1, MatchType.toSize(matchType))
+      }
+      .sortBy( - _._2)
+
+    val playerTeams = chunks
+      .map { players => players.splitAt(players.size/2) }
+      .padTo(sortedCourtTypes.length, (Iterable(), Iterable()))
```

Comment:
What are the values in the next part whe (Itenrable(), Iterable()) is matched with (courtId, courtSize)?